diff --git a/lib/prepare.js b/lib/prepare.js
index d232eb2134ab2f1f8081d3b72305caa20bf41513..1b97569f2c0c0fb190db0f8d7c0ec7201b3aba94 100644
--- a/lib/prepare.js
+++ b/lib/prepare.js
@@ -8,13 +8,9 @@ module.exports = async (npmrc, {tarballDir, pkgRoot}, {cwd, env, stdout, stderr,
   logger.log('Write version %s to package.json in %s', version, basePath);
 
   const versionResult = execa(
-    'npm',
-    ['version', version, '--userconfig', npmrc, '--no-git-tag-version', '--allow-same-version'],
-    {
-      cwd: basePath,
-      env,
-      preferLocal: true,
-    }
+    'yarn',
+    ['workspace', pkg.name, 'npm', 'publish', '--tag', distTag],
+    {cwd, env, preferLocal: true}
   );
   versionResult.stdout.pipe(stdout, {end: false});
   versionResult.stderr.pipe(stderr, {end: false});
