diff --git a/lib/publish.js b/lib/publish.js
index ff3af00789942572806e52627b5e22c8ac25d1b1..6e06e87687c93fc1d30a37f5d13740db645dc685 100644
--- a/lib/publish.js
+++ b/lib/publish.js
@@ -20,9 +20,14 @@ module.exports = async (npmrc, {npmPublish, pkgRoot}, pkg, context) => {
     const distTag = getChannel(channel);
 
     logger.log(`Publishing version ${version} to npm registry on dist-tag ${distTag}`);
+    // const result = execa(
+    //   'npm',
+    //   ['publish', basePath, '--userconfig', npmrc, '--tag', distTag, '--registry', registry],
+    //   {cwd, env, preferLocal: true}
+    // );
     const result = execa(
-      'npm',
-      ['publish', basePath, '--userconfig', npmrc, '--tag', distTag, '--registry', registry],
+      'yarn',
+      ['workspace', pkg.name, 'npm', 'publish', '--tag', distTag],
       {cwd, env, preferLocal: true}
     );
     result.stdout.pipe(stdout, {end: false});
